src/app/home-page/components/finish-modal/finish-modal.component.ts

```diff
@@ -1,31 +1,39 @@
-import { Component,OnInit } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { TrackerDurationEnum } from 'src/app/shared-module/tracker/types/tracker-duration.enum';
 import { TrackerService } from '../../../shared-module/tracker/tracker.service';
 import { TrackerEventEnum } from '../../../shared-module/tracker/types/tracker-event.enum';
 
 @Component({
   selector: 'app-finish-modal',
   templateUrl: './finish-modal.component.html',
-  styleUrls: ['./finish-modal.component.scss']
+  styleUrls: ['./finish-modal.component.scss'],
 })
+export class FinishModalComponent implements OnInit {
+  pomodoro_visibility = true;
+  break_visibility = true;
 
-export class FinishModalComponent implements OnInit{
-  visibility:boolean = true;
-  
-  constructor(
-    private tracker: TrackerService
-  ) {}
+  constructor(private tracker: TrackerService) {}
 
-  ngOnInit(): void
-  {
-    this.tracker.event.subscribe((event)=> {
-      if (event.eventType === TrackerEventEnum.finish)
+  ngOnInit(): void {
+    this.tracker.event.subscribe((event) => {
+      if (
+        event.eventType === TrackerEventEnum.finish &&
+        event.duration === TrackerDurationEnum.pomodoro
+      ) {
+        this.pomodoro_visibility = false;
+        setTimeout(() => (this.pomodoro_visibility = true), 7000);
+      } 
+      else if(event.eventType === TrackerEventEnum.finish &&
+        event.duration === TrackerDurationEnum.longBreak)
       {
-        this.visibility = false;
-        setTimeout(()=>this.visibility = true,7000)
+        this.break_visibility = false;
+        setTimeout(() => (this.break_visibility = true), 10000);
       }
-      else
+      else if(event.eventType === TrackerEventEnum.finish &&
+        event.duration === TrackerDurationEnum.shortBreak)
       {
-        this.visibility = true;
+        this.break_visibility = false;
+        setTimeout(() => (this.break_visibility = true), 10000);
       }
```

Comment:
I would refactor this code snippet using the following if-else logic:
`if (event.eventType === TrackerEventEnum.finish) {`
`  if (event.duration === TrackerDurationEnum.pomodoro) { ... }`
`  else { ... }`
`}`

Reply:
Thanks for your comment. I changed it